spec.md

```diff
@@ -0,0 +1,547 @@
+# Container Object Storage Interface (COSI)
+
+Authors:
+
+* Sidharth Mani  << @wlan0
+* Srini Brahmaroutu <<srbrahma@us.ibm.com>> (@brahmaroutu)
+
+## Notational Conventions
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119) (Bradner, S., "Key words for use in RFCs to Indicate Requirement Levels", BCP 14, RFC 2119, March 1997).
+
+The key words "unspecified", "undefined", and "implementation-defined" are to be interpreted as described in the [rationale for the C99 standard](http://www.open-std.org/jtc1/sc22/wg14/www/C99RationaleV5.10.pdf#page=18).
+
+An implementation is not compliant if it fails to satisfy one or more of the MUST, REQUIRED, or SHALL requirements for the protocols it implements.
+An implementation is compliant if it satisfies all the MUST, REQUIRED, and SHALL requirements for the protocols it implements.
+
+## Terminology
```

Comment:
Most of this terminology comes from official CSI docs. Do we need all of this in the bucket [KEP](https://github.com/kubernetes/enhancements/pull/1383)? Eg. I don't think CO, SP, RPC, Node, plugin supervisor are needed.

Reply:
I am not assuring that other CO will make sense, I can replace CO with Kubernetes, SP is the backend, I am not sure RPC, Node and plugin definitions will change much from CSI to COSI.
Sorry, I wrote this with CSI document as base, if I wasted anytime. I believe once KEP is merged we should standup 'spec' repo with this content for vendors to write plugins?

Reply:
@brahmaroutu That's a good point about having a vendor COSI driver (plugin) guideline. 
At a high level, my question about this PR is where it fits. Will it be embedded in KEP 1383, or is it intended to be stand-alone (even if it references KEP 1383)? The past couple of gRPC prs in the _yard-turkey/enhancements_ repo for KEP 1383 have been merged into the KEP. Is that the expectation with this PR, or something different? Thanks.